### PR TITLE
feat(driver): add promise reject handle

### PR DESCRIPTION
**Pull request by omega-xiao**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: framework/android/connector/driver/js/src/main/cpp/src/js_driver_jni.cc [hunk 1]
```diff
@@ -327,7 +327,13 @@ jboolean RunScriptFromUri(JNIEnv* j_env,
   };
 
   runner->PostTask(std::move(func));
-
+#ifdef JS_V8
+  auto process_handle_callback = [ctx] {
```
Comment (churchill-zhang) [inline framework/android/connector/driver/js/src/main/cpp/src/js_driver_jni.cc:331]: 不能强持有吧

File: driver/js/src/modules/timer_module.cc [hunk 2]
```diff
@@ -132,6 +135,10 @@ std::shared_ptr<hippy::napi::CtxValue> TimerModule::Start(
     if (!repeat) {
       timer_map->erase(task_id);
     }
+#ifdef JS_V8
+    auto ctx = std::static_pointer_cast<hippy::napi::V8Ctx>(scope->GetContext());
+    ctx->ProcessPromiseReject();
```
Comment (churchill-zhang) [inline driver/js/src/modules/timer_module.cc:140]: 这个不能只在这里加吧，每一个js task后都要检查

**Review by zoomchan-cxj — APPROVED** [review]

**Review by ilikethese — APPROVED** [review]

File: driver/js/lib/modules/PromiseRejectHandle.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+(function promiseRejectHandler(eventName, reason, promise) {
+  if (global.Hippy) {
+    global.Hippy.emit(eventName, {"reason" : reason, "promise" : promise});
+  } else {
+    /* eslint-disable-next-line no-console */
+    console.error(eventName, err);
```
Comment (zoomchan-cxj) [inline driver/js/lib/modules/PromiseRejectHandle.js:26]: 没有 err

**Review by zoomchan-cxj — CHANGES_REQUESTED** [review]
fix comment issue

**Review by zoomchan-cxj — CHANGES_REQUESTED** [review]
need changed

**Pull request closed without merging**